lib/clean-source-with.nix

```diff
@@ -44,6 +44,7 @@
   cleanSourceWith = { filter ? _path: _type: true, src, subDir ? "", name ? null }:
     let
       subDir' = if subDir == "" then "" else "/" + subDir;
+      subDirName = __replaceStrings ["/"] ["-"] subDir;
```

Comment:
Nit, but I find it confusing that we use `""` as the default value for `subDir`, but `null` as the default value for `name`...

Reply:
Haskell types I was thinking of `subDir :: FilePath` and `name :: Maybe String`.

I wanted to avoid the use of `null` for `subDir` and since directory names cannot be empty it seems like a natural default.  Since subDir works more or less like `src + "/" + subDir` I felt this was consistent with `ls dir` and `ls dir/`.

What I was going for with `name` is:

* `null` : build a name from the `subDir` if it can (warning if it cannot).
* `""` : Use no name regardless of `subDir` and with no warning (might be useful if you always plan to filter sub directories out).  For instance this should not produce a warning or include a `"unnamed"` in the name:

  ```
  root = cleanSourceWith { src ./.; name = ""; filter = ... };
  x = cleanSourceWith { src = root; subDir = "x"; }
  ```

Reply:
I'd have thought `subDir :: Maybe FilePath` too, since it's optional, no? But yes, I guess the empty path is a reasonable default for `subDir`, I guess I'm just surprised that you then have a bunch of checks for `subDir == ""`.

Reply:
If we used `null` as the default I think we would still need the `subDir == ""` checks (or to transform `""` to `null`).  Otherwise passing `""` would result in extra `'/'` characters in the path and the filter function would not work (we could canonicalise the path later, but I would prefer just to make it hard to have non canonical paths in the first place).